### DISCUSSION TITLE
Mounted directory with withMountedDirectory is not updated on host

### DISCUSSION THREAD

**Discussion by miroljub1995** [dagger/dagger#6688] [Help] [answered]
I'm having local dagger `v0.9.10 (registry.dagger.io/engine) darwin/arm64` and having mounted directory using `withMountedDirectory` and expected that changes to that directory are reflected in host, but no changes are made to host directory.
If that's by design, then what's the difference between `withMountedDirectory` and `withDirectory`?

**Comment by marcosnils** [comment, marked as answer]
hey there! `WithMountedDirectory` works the same way as the `RUN --mount` instruction that you see in Dockerfiles. 

It's not `bind mounting` the files in your machine to the build, what it does is mounting the files from your `build context` into your pipeline so your resulting container doesn't contain those files in the layered filesystem. If you want the files that you change in your pipeline in your host, you need to use the `Export` function. There's more info about that here: https://docs.dagger.io/421437/work-with-host-filesystem/#export-a-directory-from-a-container-to-the-host

**Reply by miroljub1995** [reply]
I see, thanks for the quick reply.
The only problem I see, also where we needed mounting, is getting test results for later use. The problem is that export to host will not be executed if test fails.

**Reply by marcosnils** [reply]
re: container exit, we have a cookbook for those cases here: https://docs.dagger.io/cookbook/#continue-using-container-after-command-execution-fails
